backend/src/layer/data/image_validation/google_api_handler.rs

```diff
@@ -18,37 +20,74 @@ use std::io::Cursor;
 ///     - calling the api<br>
 ///     - determine which images are allowed<br>
 ///     - and returning helpful error messages if an image got not accepted
+#[derive(Default)]
 pub struct GoogleApiHandler {
-    evaluation: ImageEvaluation,
-    request: ApiRequest,
+    safe_search_handler: Option<SafeSearchHandler>,
+    gemini_handler: Option<GeminiHandler>,
+}
+
+struct SafeSearchHandler {
+    evaluation: SafeSearchEvaluation,
+    request: SafeSearchRequest,
+}
+
+struct GeminiHandler {
+    evaluation: GeminiEvaluation,
+    request: GeminiRequest,
 }
 
 impl GoogleApiHandler {
     /// This method creates a new instance of the [`GoogleApiHandler`].
     /// # Params
     /// `info`<br>
-    /// This struct contains all information :) that is needed to setup this struct like
+    /// This struct contains all information :) that is needed to set up this struct like
     /// authentication information and acceptance level for the image evaluation.
     /// # Errors
-    /// It is possible that the [`ApiRequest`] creation fails, as the provided information causes failures.
+    /// It is possible that the [`SafeSearchRequest`] creation fails, as the provided information causes failures.
     /// These failures will be returned as an error.
     /// See [`crate::interface::image_validation::ImageValidationError`] for more info about the errors.
     /// # Return
     /// The mentioned [`GoogleApiHandler`] struct.
     pub fn new(info: ImageValidationInfo) -> Result<Self> {
-        Ok(Self {
-            evaluation: ImageEvaluation::new(info.acceptance),
-            request: ApiRequest::new(&info.service_account_info, info.project_id)?,
-        })
+        let mut handler = Self::default();
+        if let Some(info) = info.safe_search_info {
+            handler.safe_search_handler = Some(SafeSearchHandler {
+                evaluation: SafeSearchEvaluation::new(info.acceptance),
+                request: SafeSearchRequest::new(&info.service_account_info, info.project_id)?,
+            });
+        }
+        if let Some(info) = info.gemini_info {
+            handler.gemini_handler = Some(GeminiHandler {
+                evaluation: GeminiEvaluation::default(),
+                request: GeminiRequest::new(info.gemini_api_key, &info.gemini_text_request),
+            });
+        }
+        Ok(handler)
```

Comment:
That, again, I find rather unreadable (I don't like mut variables when not necessary😏).
```suggestion
        Ok(Self {
            safe_search_info = if let Some(info) = info.safe_search_info {Some(SafeSearchHandler {
                evaluation: SafeSearchEvaluation::new(info.acceptance),
                request: SafeSearchRequest::new(&info.service_account_info, info.project_id)?,
            } else {None},
            gemini_handler = info.gemini_handler.map(|info| GeminiHandler {
                evaluation: GeminiEvaluation::default(),
                request: GeminiRequest::new(info.gemini_api_key, &info.gemini_text_request),
            })
        })
```
Maybe the first part is possible even cleaner although of the `?`.
This makes it more explicit which variables Self contains. If we add new ones, they cannot simpy be forgotten to be initialized.